exercises/concept/poetry-club-door-policy/door-policy.spec.js

```diff
@@ -7,239 +9,117 @@ import {
 
-Michael Lockwood
-`.trim()
-);
-
-const SHIRE_HORSE_WITH_SPACES = new Poem(
-  `
-Stands so high  
-Huge hooves too      
-Impatiently waits for 
-Reins and harness  
+//Shire Horse, by Michael Lockwood
+const SHIRE_HORSE_WITH_SPACES = `
+Stands so high
+Huge hooves too
+Impatiently waits for
+Reins and harness
 Eager to leave
+`.split('\n');
 
-Michael Lockwood
-`
-);
-
-const SUMMER = new Poem(`
+//Summer, by John Albert Caballero
+const SUMMER = `
 Sunshine warming my toes,
 Underwater fun with my friends.
 Making homemade ice cream on the porch,
 Many long nights catching fireflies.
 Early morning walks to the creek,
 Reveling in the freedom of lazy days.
+`.split('\n');
 
-John Albert Caballero
-`);
-
-const SOPHIA = new Poem(
-  `
+//Sophia, by John Albert Caballero
+const SOPHIA = `
 Serene, calming quality
 Organized, you always have it together
 Picturesque, strikingly beautiful
 Honest, so genuine
 Imaginative, a creative mind
 Alluring, so attractive
+`
+  .trim()
+  .split('\n');
 
-John Albert Caballero
-`.trim()
-);
-
-const CODE_WORK = new Poem(
-  `
+//Code Work, by Derk-Jan Karrenbeld
+const CODE_WORK = `
 Compilers intensily bestow
 On commencing without ego
 Different processes ajar
 Exit with zero quick
-
-Derk-Jan Karrenbeld
-`.trim()
-);
+`
+  .trim()
+  .split('\n');
 
 describe('strings', () => {
-  describe('front door', () => {
-    const ShireGuard = new FrontDoorGuard(SHIRE_HORSE);
-
-    test('it outputs a character per line', () => {
-      const key = ShireGuard.recite();
+  describe('front door response', () => {
+    test('should output a character per line', () => {
+      const key = recite(SHIRE_HORSE, frontDoorPatron);
       expect(key.length).toBe(SHIRE_HORSE.length);
     });
 
-    test('it outputs takes the first characters', () => {
-      const key = ShireGuard.recite();
-      expect(key.toUpperCase()).toBe(SHIRE_HORSE.acrostic.toUpperCase());
+    test('should take the first character of each line', () => {
+      const key = recite(SHIRE_HORSE, frontDoorPatron);
+      const expectedFirstLetters = ['S', 'H', 'I', 'R', 'E'];
+      expect(key).toEqual(expectedFirstLetters);
     });
 
-    test('it generates the correct password', () => {
-      ShireGuard.assert();
+    test('should generate the correct password', () => {
+      expect(frontDoorPassword('SHIRE')).toBe('Shire');
     });
 
-    const FRONT_DOOR_CASES = {
-      SUMMER,
-      SOPHIA,
-      CODE_WORK,
-    };
-
-    Object.keys(FRONT_DOOR_CASES).forEach((name) => {
-      const poem = FRONT_DOOR_CASES[name];
-      const guard = new FrontDoorGuard(poem);
-
-      test(`frontDoorPassword(${name})`, () => {
-        guard.assert();
-      });
-    });
+    test.each([
+      { poem: SUMMER, firstLetters: 'SUMMER', password: 'Summer' },
+      { poem: SOPHIA, firstLetters: 'SOPHIA', password: 'Sophia' },
+      { poem: CODE_WORK, firstLetters: 'CODE', password: 'Code' },
+    ])(
+      'should be correct for $password',
```

Comment:
`$password` as text does not seem very descriptive, maybe add a name property to the test objects above that contains the name of the poem, that you can then use here

Reply:
`test.each` _should_ work well but it doesn't. Can you make them three separate tests? Sorry! Our test-runner isn't great yet.

Reply:
`$password` should have been outputting the password from the each block, but wasn't in this version of jest. Tests now split out.